dm/master/election_test.go

```diff
@@ -32,7 +32,7 @@ var _ = check.Suite(&testElectionSuite{})
 type testElectionSuite struct{}
 
 func (t *testElectionSuite) TestFailToStartLeader(c *check.C) {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
```

Comment:
Even `WithTimeout`, this should work, doesn't it?🤔

Reply:
it will failed when ctx deadline exceeded

`[2021-09-10T02:40:21.661Z] ... value *terror.Error = [code=11108:class=functional:scope=internal:level=medium], Message: fail to get leader ID, RawCause: context deadline exceeded ("[code=11108:class=functional:scope=internal:level=medium], Message: fail to get leader ID, RawCause: context deadline exceeded")`

Reply:
Maybe you can change your title to `test: fix ctx timeout in advance before test success in ut`